junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -29,6 +29,18 @@ import kotlin.contracts.contract
 fun fail(message: String?, throwable: Throwable? = null): Nothing =
     Assertions.fail<Nothing>(message, throwable)
 
+/**
+ * @see Assertions.fail
+ */
+@JvmName("fail_nonNullableLambda")
+fun fail(message: () -> String): Nothing {
+    contract {
+        callsInPlace(message)
```

Comment:
> A function declaring the callsInPlace effect must be inline. -- https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.contracts/-contract-builder/calls-in-place.html

hmm

Reply:
The question regarding inline functions has been already answered: https://youtrack.jetbrains.com/issue/KT-57153